Sources/BBCodeParser.php

```diff
@@ -3745,7 +3748,7 @@ protected function detectTag(string $tag_character): array
 			}
 
 			// A test validation?
-			if (isset($possible['test']) && preg_match('~^' . $possible['test'] . '~', substr($this->message, $this->pos + 1 + $pt_strlen + 1)) === 0) {
+			if (isset($possible['test']) && preg_match('~^' . $possible['test'] . '~', substr($this->message, (int) ($this->pos + 1 + $pt_strlen + 1))) === 0) {
```

Comment:
Casting the result of the expression to int is unnecessary. The result of the arithmetic operation will already be an integer.